OpenRA.Mods.Common/Traits/Render/WithMoveAnimation.cs

```diff
@@ -14,44 +14,49 @@
 
 namespace OpenRA.Mods.Common.Traits.Render
 {
-	public class WithMoveAnimationInfo : ITraitInfo, Requires<WithSpriteBodyInfo>, Requires<IMoveInfo>
+	public class WithMoveAnimationInfo : ConditionalTraitInfo, Requires<WithSpriteBodyInfo>, Requires<IMoveInfo>
 	{
 		[Desc("Displayed while moving.")]
 		[SequenceReference] public readonly string MoveSequence = "move";
 
 		[Desc("Which sprite body to modify.")]
-		public readonly string[] BodyNames = { "body" };
+		public readonly string Body = "body";
 
-		public object Create(ActorInitializer init) { return new WithMoveAnimation(init, this); }
+		public override object Create(ActorInitializer init) { return new WithMoveAnimation(init, this); }
+
+		public override void RulesetLoaded(Ruleset rules, ActorInfo ai)
+		{
+			var matches = ai.TraitInfos<WithSpriteBodyInfo>().Count(w => Body == w.Name);
```

Comment:
This is consistent with WithAttackAnimation. Below we use First(w => w.Info.Name == Info.Body). Not important, but couldn't we have a style rule that for such simple predicates we start with property of parameter? so Count(w => w.Name == body).